src/main/rx/spektrum.c

```diff
@@ -150,8 +150,16 @@ static uint8_t spektrumFrameStatus(rxRuntimeConfig_t *rxRuntimeConfig)
         }
 
 #if defined(USE_TELEMETRY) && defined(USE_TELEMETRY_SRXL)
-        if (srxlEnabled && telemetryBufLen && (spekFrame[2] & 0x80) == 0) {
-            telemetryFrameRequestedUs = currentTimeUs;
+        if (srxlEnabled) {
+            if (telemetryBufLen) {
+                if ((spekFrame[2] & 0x80) == 0) {
+                    telemetryFrameRequestedUs = currentTimeUs;
+                }
+            }
+            else {
+                // Trigger tm data collection if buffer is empty.
+                srxlCollectTelemetryNow();
```

Comment:
This is just plain wrong. Anything that is not a check for the received data should not be done in the frame check function.

@borisbstyle: Are you going to fix this, or should this be reverted? Also, can you please stick to the rule of leaving pull requests open for at least a day, particularly if they contain changes from contributors that have proven in the past that they have no interest in contributing code that conforms with Betaflight standards?